src/richie/plugins/lti_consumer/cms_plugins.py

```diff
@@ -0,0 +1,59 @@
+"""
+LTI consumer CMS plugin
+"""
+from django.utils.translation import gettext_lazy as _
+
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+
+from richie.apps.core.defaults import PLUGINS_GROUP
+
+from .models import LTIConsumer
+
+
+@plugin_pool.register_plugin
+class LTIConsumerPlugin(CMSPluginBase):
+    """
+    A plugin to add plain text.
+    """
+
+    allow_children = False
+    cache = True
+    disable_child_plugins = True
+    fieldsets = ((None, {"fields": ["url", "course"]}),)
+    model = LTIConsumer
+    module = PLUGINS_GROUP
+    name = _("LTI consumer")
+    render_template = "richie/lti_consumer/lti_consumer.html"
+
+    # def get_form(self, request, obj=None, change=False, **kwargs):
+    #     """
+    #     Add a max length validator based on what is configured in settings for the
+    #     current placeholder.
+    #     """
+    #     form = super().get_form(request, obj=obj, change=change, **kwargs)
+    #
+    #     placeholder_id = request.GET.get("placeholder_id")
+    #     if not placeholder_id and not obj:
+    #         return form
+    #
+    #     if placeholder_id:
+    #         placeholder = Placeholder.objects.only("slot").get(id=placeholder_id)
+    #     else:
+    #         placeholder = obj.placeholder
+    #
+    #     max_length = getattr(settings, "RICHIE_PLAINTEXT_MAXLENGTH", {}).get(
+    #         placeholder.slot
+    #     )
+    #     if max_length:
+    #         form.base_fields["body"].validators.append(MaxLengthValidator(max_length))
+    #
+    #     return form
+    #
+    def render(self, context, instance, placeholder):
+        """
+        Build plugin context passed to its template to perform rendering
+        """
+        context = super().render(context, instance, placeholder)
+        context["url"] = instance.url
```

Comment:
or pass the whole instance?

Reply:
Instance is already passed in CMSPluginBase.render()

```python
    def render(self, context, instance, placeholder):
        context['instance'] = instance
        context['placeholder'] = placeholder
        return context
```
In fact overriding render is useless for now.
But do you prefer to have it explicitly defined at the plugin level ?